test/unit/typechecking.spec.ts

```diff
@@ -140,3 +140,52 @@ test("instanceof Symbol.hasInstance", () => {
 
     expect(result).toBe(true);
 });
+
+test.each([
+    { expression: "{}", operator: "===", compareTo: "object", expectResult: "TRUE" },
+    { expression: "{}", operator: "!==", compareTo: "object", expectResult: "FALSE" },
+    { expression: "{}", operator: "==", compareTo: "object", expectResult: "TRUE" },
+    { expression: "{}", operator: "!=", compareTo: "object", expectResult: "FALSE" },
+    { expression: "{}", operator: "<=", compareTo: "object", expectResult: "TRUE" },
+    { expression: "{}", operator: "<", compareTo: "object", expectResult: "FALSE" },
+    { expression: "undefined", operator: "===", compareTo: "undefined", expectResult: "TRUE" },
+    { expression: "() => {}", operator: "===", compareTo: "function", expectResult: "TRUE" },
+    { expression: "1", operator: "===", compareTo: "number", expectResult: "TRUE" },
+    { expression: "true", operator: "===", compareTo: "boolean", expectResult: "TRUE" },
+    { expression: `"foo"`, operator: "===", compareTo: "string", expectResult: "TRUE" },
+])("typeof literal comparison (%p)", ({ expression, operator, compareTo, expectResult }) => {
+    const code = `
+        let val = ${expression};
+        if (typeof val ${operator} "${compareTo}") {
+            return "TRUE";
```

Comment:
This looks a little odd, any reason you don't just `return typeof val ${operator} "${compareTo}"`?